### DISCUSSION TITLE
xcom saved value of type numpy.bool_ is changed from False to True

### DISCUSSION THREAD

**Discussion by kkeiichi** [apache/airflow#42968] [General]
### Apache Airflow version

2.10.2

### What happened?

A `False` value of `numpy.bool_` type pushed to xcom is changed to `True` on pull. A `True` `numpy.bool_` value stays `True`.
A regular Python `bool` value works fine.

### What you think should happen instead?

`False` value of `numpy.bool_` type pushed to xcom should stay `False`.

### How to reproduce

    from datetime import datetime
    import logging
    from airflow.models.dag import DAG
    from airflow.operators.python import PythonOperator
    import numpy
    
    
    def xcom_test(ti):
        value_in = numpy.bool_(False)
        ti.xcom_push(key="test_value", value=value_in)
        value_out = ti.xcom_pull(key="test_value", task_ids="xcom_test")
        logging.info("In: " + str(value_in))
        logging.info("Out: " + str(value_out))
    
    
    with DAG(
            "test",
            start_date=datetime(2024, 8, 27),
            schedule=None,
            catchup=False,
            max_active_runs=1
    ) as dag:
        PythonOperator(
            task_id="xcom_test",
            python_callable=xcom_test
        )

Logs:

    [2024-10-09, 18:56:12 CEST] {test.py:12} INFO - In: False
    [2024-10-09, 18:56:12 CEST] {test.py:13} INFO - Out: True

### Operating System

Ubuntu 24.04

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by gopidesupavan** [comment]
hm, strange, looking into that.

**Comment by gopidesupavan** [comment]
Alright the value numpy type not supported currently in xcom. any reason your using xcom for numpy types?

**Comment by kkeiichi** [comment]
No reason. It just seemed odd to me. I intended to use a regular bool, but accidentally used numpy. I was surprised by the result.

**Comment by potiuk** [comment]
Converted into discussion. Maybe a good idea that someone add a bit more explanation - but we do not need issue for it to avoid the surprises - https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html#xcoms contains information that Xcom should be serialiable. But if you have indea @kkeiichi on how to make it more obvious from the docs or error message - feel free to open PR for it.
